src/pools/ID3MPool.sol

```diff
@@ -16,15 +16,82 @@
 
 pragma solidity 0.6.12;
 
+/**
+    @title D3M Pool Interface
+    @notice Pool contracts are the contract that the Hub uses to standardize interactions with external Pools
+    @dev This contract will hold any balance provided by the external pool as well as the balance in the Vat.
+    This contract aims to use EIP-4626 guidelines for assets/shares/maxWithdraw etc.
+*/
 interface ID3MPool {
+    /**
+        @notice Function that will deposit assets (Dai) in the external pool.
+        @dev If the external pool requires a different amount be passed in the
+        conversion should occur here as the Hub passes Dai [wad] amounts.
+        @param amt amount in asset (Dai) terms that we want to deposit
+    */
     function deposit(uint256 amt) external;
```

Comment:
In other places you write msg.sender must be authorized, but not here.
Relevant also for `withdraw()`, `transfer()`, `transferAll()`.